src/index.tsx

```diff
@@ -138,11 +150,17 @@ const toEffectObject = <
   TEvent extends EventObject,
   TEffect extends EffectObject<TState, TEvent>
 >(
-  effect: TEffect | EffectFunction<TState, TEvent>,
-  effectsMap?: EffectsMap<TState, TEvent>
+  effect: TEffect | EffectFunction<TState, TEvent, TEffect>,
+  effectsMap?: EffectsMap<TState, TEvent, TEffect>
 ): TEffect => {
   const type = typeof effect === 'function' ? effect.name : effect.type;
-  const customExec = effectsMap ? effectsMap[type] : undefined;
+  const customExec = effectsMap
+    ? ((effectsMap as any)[type as string] as EffectFunction<
```

Comment:
This is the one place where I messed up - I am not quite sure why the new type messes with the expression `effectsMap[type]` - in theory it should all fit together